### DISCUSSION TITLE
[MongoDB] `orderBy: { updatedAt: 'desc' }` (`DateTime`) returns same documents when using `skip` and `take` and some documents having same `updatedAt` field

### DISCUSSION THREAD

**Discussion by bn-piar** [prisma/prisma#16592] [Q&A] [answered]
Hi guys. 

I'm facing a problem with pagination and sorting by `DateTime` field when many documents have same `DateTime` value: it happens that I'm getting the same documents on different pages. 

For example, 
```ts
const data = await prisma.data.findMany({ orderBy: { updatedAt: 'desc' }, take: 5, skip: 0 * 5 });
/* DATA */
/*
[
    {
        "id": "6389c2894dae55666718c04f"
    },
    {
        "id": "6389c2894dae55666718b909"
    },
    {
        "id": "6389c2894dae55666718f020"
    },
    {
        "id": "6389c2894dae55666718e10d"
    },
    {
        "id": "6389c2894dae55666718f451"
    }
]
*/
const data1 = prisma.data.findMany({ orderBy: { updatedAt: 'desc' }, take: 5, skip: 1 * 5 });
/* DATA1 */
/*
[
    {
        "id": "6389c2894dae55666718b909"
    },
    {
        "id": "6389c2894dae55666718f020"
    },
    {
        "id": "6389c2894dae55666718e10d"
    },
    {
        "id": "6389c2894dae55666718f451"
    },
    {
        "id": "6389c2894dae55666718ba44"
    }
]
*/
```
So `data1` ( page 1) includes some documents from page 0. 

How can I avoid it?

**Comment by nurul3101** [comment, marked as answer]
Hello @bn-piar 👋 

I would recommend you add a fallback in orderBy to get a stable sort and avoid getting the same documents.
You can do something like this:
```typescript
const data = await prisma.data.findMany({ orderBy: [{ updatedAt: 'desc' }, { id: 'asc' }], take: 5, skip: 0 * 5 });
``` 

[Documentation Reference](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#sort-user-by-multiple-fields---email-and-role)

Let me know if this works for you.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
